kolena/_experimental/dataset/common.py

```diff
@@ -58,4 +58,6 @@ def validate_dataframe_ids(df: pd.DataFrame, id_fields: List[str]) -> None:
             raise InputValidationError(
                 f"invalid id_fields: field '{id_field}' does not exist in dataframe",
             )
+        elif df[id_field].dtype == object and not isinstance(df.iloc[0][id_field], str):
```

Comment:
Just double checking: types like pandas categorical or timestamp would also fall here, i.e. `object`?

Reply:
datetime will be it's own dtype
ref: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.dtypes.html

Reply:
Confirming these won't fall under object
```
>>> pd.Series(["a", "b", "c"], dtype="category").dtype == object
False
>>> pd.Series(pd.date_range("2000", freq="D", periods=3)).dtype == object
False
```